provenance.md

```diff
@@ -1,50 +1,113 @@
+The workflow provenance is stored in [PROV-XML](https://www.w3.org/TR/prov-xml/). 
+
+### Types
+All Activities, Entities belonging to the OCR-D workflow have the same namespace.
+#### Namespace
+<dl>
+  <dt><strong>Prefix</strong></dt>
+  <dd>ocrd</dd>
+  <dt><strong>Namespace</strong></dt>
+  <dd>http://www.ocr-d.de</dd>
+</dl>
+
+Type | Data Type | Description
+-------- | -------- | --------
+Entity   | ocrd:mets | Filename  of METS file
+Entity   | ocrd:mets_referencedFile | ID of the file referenced inside METS. 
+Entity   | ocrd:parameter_file   | Content of the parameter file.
+Activity   | ocrd:module | Module that was executed 
+Activity   | ocrd:workflow | Workflow that was executed
+
 
 ## Content 
 Only the following information is stored for provenance:
-(a) General data
-1. Workflow engine (including version)
-2. METS file
-(b) Module Projects
-1. ID of the input file(s)
-2. Content of parameter.json (optional)
-3. Module 
-  i) unique-id
-  ii) start date
-  iii) end date
-  iiii) (ocrd-tool.json)?
-4. ID of output files
-5. Log level (?) this is only used for debugging
-6. StdOut(?) this is only used for debugging
-7. StdErr(?) this is only used for debugging
+(a) **General data**
+1. Workflow engine 
+  - Label including version
+  - Start date
+  - End date
+
+(b) **Module Projects**
+2. Module 
+  - Label (e.g.: ocrd-kraken-binarize_Version 0.1.0, ocrd/core 1.0.0)
+  - Start date
+  - End date
+3. METS file before executing module
+4. METS file after executing module
+5. ID of the input file(s)
+6. ID of output file(s)
+7. Content of parameter.json (optional)
 
 ### Input/Output
 All files listed in METS should be referenced in provenance via their file ID.
 File ID MAY be linked to its location. The location may be replaced due to 
 different uses:
 1. local files
 2. external files
-Files not listed in METS should be linked to their content in provenance.
+Files not listed in METS should be linked to their content in provenance. (e.g.: parameter.json)
 
 ### Ingest Workspace to OCR-D Repositorium
-During the ingest into repository/LTA, the entire provenance must be stored in a graph to make the provenance searchable.
+At least before ingesting into repository/LTA, the entire provenance must be stored in one file (metadata/ocrd_provenance.xml) to make the provenance searchable.
 Therfore all the provenance files are merged into one big file.
-This file replaces all files stored in subfolder 'metadata/provenance'
+This file replaces all provenance files stored in subfolder 'metadata'
+
+## Example
+The file structure could look like this after a workflow with 4 steps has been executed.
+```
+metadata/
+   |
+   +-- mets.xml.'workflowid'_0000
+   |
+   +-- mets.xml.'workflowid'_0001
+   |
+   +-- mets.xml.'workflowid'_0002
+   |
+   +-- mets.xml.'workflowid'_0003
+   |
+   +-- mets.xml.'workflowid'_0004
+   |
+   +-- ocrd_provenance.xml
+   |
+   +-- provenance_'workflowid'.xml (optional)
+```
```

Comment:
This example is staggeringly incomplete. It would be effective as an example only with the actual _content_ of the named provenance XML files.

Reply:
It's just a little example to show where the files are stored and how the structure looks like. The schema of the provenance XML files are defined by https://www.w3.org/TR/2013/NOTE-prov-xml-20130430/. It's not helpful to show unreadable XML.

Reply:
Unreadable XML?

The structure is somewhat informative in itself, granted. But you have different kinds of files here (with different schemas?), and so far no schema was given at all, let alone an instance. If this is meant to be illustrative, you _must_ add an actual example (regardless of how artificial _or_ repetetive it may seem).